flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java

```diff
@@ -70,6 +71,38 @@
 	 */
 	private long maxIdleStateRetentionTime = 0L;
 
+	/**
+	 * A configuration object to hold all key/value configuration.
+	 */
+	private Configuration configuration = new Configuration();
+
+	/**
+	 * Returns all key/value configuration.
+	 */
+	public Configuration getConfiguration() {
+		return configuration;
+	}
+
+	/**
+	 * Adds the given key/value configuration.
+	 *
+	 * @param configuration key/value configuration to adds
+	 */
+	public void addConfiguration(Configuration configuration) {
+		Preconditions.checkNotNull(configuration);
+		this.configuration.addAll(configuration);
+	}
+
+	/**
+	 * Adds the given key/value pair.
+	 *
+	 * @param key the key of the key/value pair to be added
+	 * @param value the value of the key/value pair to be added
+	 */
+	public void setConf(String key, String value) {
```

Comment:
Do we need this method? People can just use`tableConfig.getConfiguration()` and have a variety of setters for int, boolean, etc. available. Otherwise I see PRs coming that want to add additional methods and basically mirror `Configuration`.